rewrite-maven/src/main/java/org/openrewrite/maven/AddPlugin.java

```diff
@@ -90,6 +104,15 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
 
     private class AddPluginVisitor extends MavenIsoVisitor<ExecutionContext> {
 
+        @Override
+        public boolean isAcceptable(SourceFile sourceFile, ExecutionContext executionContext) {
+            if (filePattern != null) {
+                return PathUtils.matchesGlob(sourceFile.getSourcePath(), filePattern);
+            } else {
+                return sourceFile instanceof Xml.Document;
+            }
```

Comment:
You'll also want to call out to the super implementation defined here
https://github.com/openrewrite/rewrite/blob/de8bc15be8847027e197ad4ef73df528cc5a3f7e/rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java#L55-L57

Otherwise we risk running the visitor against any matching file, or xml document, even those that are not Maven pom files.

Reply:
Thanks, I commited a fix for this